frontend/src/views/Org/SshCaPage/components/SshCertificateModal.tsx

```diff
@@ -0,0 +1,350 @@
+        type: "error"
+      });
+    }
+  };
+
+  return (
+    <Modal
+      isOpen={popUp?.sshCertificate?.isOpen}
+      onOpenChange={(isOpen) => {
+        handlePopUpToggle("sshCertificate", isOpen);
+        reset();
+        setCertificateDetails(null);
+      }}
+    >
+      <ModalContent title="Issue SSH Certificate">
+        {!certificateDetails ? (
+          <form onSubmit={handleSubmit(onFormSubmit)}>
+            <Controller
+              control={control}
+              name="templateName"
+              defaultValue=""
+              render={({ field: { onChange, ...field }, fieldState: { error } }) => (
+                <FormControl
+                  label="Certificate Template"
+                  errorText={error?.message}
+                  isError={Boolean(error)}
+                  isRequired
+                >
+                  <Select
+                    defaultValue={field.value}
+                    {...field}
+                    onValueChange={(e) => onChange(e)}
+                    className="w-full"
+                    isDisabled={Boolean(popUpData?.sshCaId)}
+                  >
+                    {(templatesData?.certificateTemplates || []).map(({ id, name }) => (
+                      <SelectItem value={name} key={`ssh-cert-template-${id}`}>
+                        {name}
+                      </SelectItem>
+                    ))}
+                  </Select>
+                </FormControl>
+              )}
+            />
+            <FormControl label="Operation">
+              <Select
+                defaultValue="issue-ssh-creds"
+                value={operation}
+                onValueChange={(e) => setOperation(e as SshCertificateOperation)}
+                className="w-full"
+              >
+                <SelectItem value="sign-ssh-key" key="sign-ssh-key">
+                  Sign SSH Key
+                </SelectItem>
+                <SelectItem value="issue-ssh-creds" key="issue-ssh-creds">
+                  Issue SSH Credentials
+                </SelectItem>
+              </Select>
+            </FormControl>
+            <Controller
+              control={control}
+              name="certType"
+              defaultValue={SshCertType.USER}
+              render={({ field: { onChange, ...field }, fieldState: { error } }) => (
+                <FormControl
+                  label="Certificate Type"
+                  errorText={error?.message}
+                  isError={Boolean(error)}
+                >
+                  <Select
+                    defaultValue={field.value}
+                    {...field}
+                    onValueChange={(e) => onChange(e)}
+                    className="w-full"
+                  >
+                    <SelectItem value={SshCertType.USER}>User</SelectItem>
+                    <SelectItem value={SshCertType.HOST}>Host</SelectItem>
+                  </Select>
+                </FormControl>
```

Comment:
can we filter the options based on the settings configured for the template? like I shouldn't be seeing Host if it's not allowed
<img width="680" alt="image" src="https://github.com/user-attachments/assets/6a75bc90-3d9d-4d6d-b5a3-10a3969959ed">